### PR TITLE
Don't require pygraphviz for regenerate.sh

### LINKED ISSUE #17336: regenerate.sh should not require pygraphviz to be installed to run
```
(/Users/ezyang/Dev/pytorch-tmp-env) macbook-pro-116:.circleci ezyang$ ./regenerate.sh 
+ ./generate-config-yml.py
Traceback (most recent call last):
  File "./generate-config-yml.py", line 14, in <module>
    import binary_build_definitions
  File "/Users/ezyang/Dev/pytorch-tmp/.circleci/binary_build_definitions.py", line 3, in <module>
    import conf_tree
  File "/Users/ezyang/Dev/pytorch-tmp/.circleci/conf_tree.py", line 4, in <module>
    from pygraphviz import AGraph
ModuleNotFoundError: No module named 'pygraphviz'
```

### PR DESCRIPTION
**Pull request by kostmo**
closes #17336

Do not overwrite config.yml if script throws an error

### CODE REVIEW

**Comment by kostmo** [comment]
There are some minor, cosmetic-only changes to .circleci/config.yml to simplify code.

File: .circleci/cimodel/data/caffe2_build_definitions.py [hunk 1]
```diff
@@ -126,33 +126,24 @@ def gen_yaml_tree(self):
...
-
-        if self.is_build_only():
-            if not self.distro.name == "macos":
-                tuples.append(("BUILD_ONLY", miniutils.quote("1")))
-
-        # TODO: not sure we need the distinction between system and homebrew anymore. Our python handling in cmake
-        #  and setuptools is more robust now than when we first had these.
         if self.distro.name == "macos":
+            # TODO: not sure we need the distinction between system and homebrew anymore. Our python handling in cmake
+            #  and setuptools is more robust now than when we first had these.
```
Comment (pjh5) [inline .circleci/cimodel/data/caffe2_build_definitions.py]: We shouldn't need this distinction anymore

File: .circleci/cimodel/data/caffe2_build_definitions.py [hunk 2]
```diff
@@ -126,33 +126,24 @@ def gen_yaml_tree(self):
             tuples.append(("BUILD_IOS", miniutils.quote("1")))
 
         if self.phase == "test":
-            use_cuda_docker = self.compiler.name == "cuda"
-            if use_cuda_docker:
+            if self.compiler.name == "cuda":
```
Comment (pjh5) [inline .circleci/cimodel/data/caffe2_build_definitions.py]: Eventually change this so cuda is not considered a compiler

**Review by pjh5 — APPROVED** [review]

**Comment by ezyang** [comment]
It would be easier to review this PR if the cosmetic changes were stacked, ala https://github.com/ezyang/ghstack ;)

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**